### ISSUE TITLE
Reporter to report parent stage and pipeline of tasks and stages

### ISSUE THREAD

**Issue by mturilli** [radical-cybertools/radical.entk#306]
The default output of EnTK is something like:
```
EnTK session: re.session.Maras-MBP.furious.lan.mturilli.017942.0009
Creating AppManager                                                           ok
Validating and assigning resource manager                                     ok
Setting up RabbitMQ system                                                    ok
All components created
Update: Pipeline pipeline.0000 in state SCHEDULING
Update: Stage stage.0000 in state SCHEDULING
Update: Task task.0000 in state SCHEDULING
Update: Pipeline pipeline.0001 in state SCHEDULING
Update: Stage stage.0002 in state SCHEDULING
Update: Task task.0007 in state SCHEDULING
Update: Task task.0008 in state SCHEDULING
Update: Task task.0000 in state SCHEDULED
```
When prototyping a workflow skeleton, it is difficult to infer the workflow's structure from that output. Something like the following would make it easier:
```
Update: Task task.0000, Stage stage.0000, Pipelines pipeline.0000 in state SUBMITTING
```

**Comment by vivek-bala** [comment]
I think we should capture the information that specified state is for a task/stage/pipeline.
My suggestion:

```
# For tasks
Update: Task task.0000 of Stage stage.0000 of Pipeline pipeline.0000 in state SUBMITTING
# For stages
Update: Stage stage.0000 of Pipeline pipeline.0000 in state SCHEDULING
# For pipelines
Update: Pipeline pipeline.0000 in state SCHEDULING
```

**Comment by mturilli** [comment]
I tested #307 and it works. Unfortunately, it is less useful than I thought. I assigned names to pipelines and stages as per API spec but they are not reported by the reporter. Any insight in how workflow structure/priority checking is usually done in EnTK when testing a new workflow app?

**Comment by andre-merzky** [comment]
There might not exist a single good solution, and I would caution against over-engineering this.  Having said that, we could switch behavior with an env variable and report names instead of UIDs, to get something like:
```
Update: pipeline.0009.stage.0029.task.0029 state: DEQUEUED
Update: pipeline.0009.stage.0029.task.0029 state: DONE
Update: pipeline.0009.stage.0029 state: DONE
Update: pipeline.0009 state: DONE
```

switched to 

```
Update: pipeline.0008.md_stage.task.0028 state: DEQUEUED
Update: pipeline.0008.md_stage.task.0028 state: DONE
Update: pipeline.0008.md_stage state: DONE
Update: pipeline.0008 state: DONE
Update: pipeline.0009.ex_stage.task.0029 state: DEQUEUED
Update: pipeline.0009.ex_stage.task.0029 state: DONE
Update: pipeline.0009.ex_stage state: DONE
Update: pipeline.0009 state: DONE
```

(here is assumed that only the stage has an application specified name).
If that is what you would like to see let me know, happy to give this a shot.  FWIW, #307 does work ok for me in the repex use case, and I would  not mind keeping it, even if it turns out being imperfect.

**Comment by vivek-bala** [comment]
PST objects will only have names if the user assigned them one. All objects will uids though. The current implementation in #307 looks pretty good to me. 

@andre-merzky do you think we should have another environment variable to control that? I think we should go with names if the user has specified them or backtrack to uids.

```python
# For a task

@property
def luid(self):
    pipe_id = self.parent_pipeline['name'] if self.parent_pipeline['name'] else self.parent_pipeline['uid']
    stage_id = self.parent_stage['name'] if self.parent_stage['name'] else self.parent_stage['uid']
    task_id = self.name if self.name else self.uid
    return '%s.%s.%s' % (pipe_id, stage_id, task_id)
```

Makes the function a bit longer, but I would favor this over another environment variable that we have to tell the user.

**Comment by mturilli** [comment]
@andre-merzky I do not like using env variables for configuring our tools. I understand it is the standard approach in the unix ecosystem but the past years showed us that: most of our users are not as accustomed to this as we are; and we are not good at reliably document all the variables.  In OOP I like creating objects (Reporter in this case as done with WMS few years ago) and configure the properties of that object to behave as I please. Unfortunately, I think this may seem over engineering as per your warning :)

@vivek-bala I assigned names to both pipelines and stages by setting their `name` property as per API. Once done, I did indeed check whether those names were used in the reporter so this seems to confirm your assumption that if a user sets the name property, her expectation is to see/use those names.

**Comment by andre-merzky** [comment]
I don't mind env variables and *not* telling them to users if the default is sane and useful ;-)  Having said that, either one is acceptable.  I personally prefer the UID version as UIDs are more canonical and uniform than names (a user can give the same name to different things which would be greatly confusing the purpose here) - but I can see why users want to see recognizable strings...

**Comment by vivek-bala** [comment]
Thanks for the PR Matteo. Released in 0.7.14.
